File: app/src/main/java/net/hearnsoft/gensokyoradio/trd/utils/AudioSessionManager.java
package net.hearnsoft.gensokyoradio.trd.utils;

import android.content.Context;
import android.media.AudioManager;

public class AudioSessionManager {


    private Integer audioSessionId;

    private static AudioSessionManager audioSessionManager;

    public static AudioSessionManager getInstance() {
        if (audioSessionManager == null) {
            audioSessionManager = new AudioSessionManager();
        }
        return audioSessionManager;
    }

    public AudioSessionManager() {}

    public void generateAudioSessionId(Context context) {
        AudioManager manager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
        audioSessionId = manager.generateAudioSessionId();
    }

    public int getAudioSessionId() {
        return audioSessionId;
    }

}

File: app/src/main/java/net/hearnsoft/gr3rd/compose/utils/AudioSessionManager.kt
package net.hearnsoft.gr3rd.compose.utils

import android.content.Context
import android.media.AudioManager

object AudioSessionManager {
    private var audioSessionId: Int? = null

    fun generateAudioSessionId(context: Context): Int {
        if (audioSessionId == null) {
            val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
            audioSessionId = audioManager.generateAudioSessionId()
        }
        return audioSessionId!!
    }

    fun getAudioSessionId(): Int? = audioSessionId

    fun getAudioSessionIdOrDefault(): Int = audioSessionId ?: AudioManager.AUDIO_SESSION_ID_GENERATE

    fun reset() {
        audioSessionId = null
    }
}